transport.go

```diff
@@ -218,24 +217,20 @@ type webhookValidator struct {
 	now func() time.Time
 }
 
-func (wv webhookValidator) setWebhookDefaults(register any, requestOriginHost string) {
-	switch r := register.(type) {
-	case webhook.RegistrationV1:
-		if len(r.Matcher.DeviceID) == 0 {
-			r.Matcher.DeviceID = []string{".*"} // match anything
-		}
-		if r.Until.IsZero() {
-			r.Until = wv.now().Add(time.Duration(r.Duration))
-		}
-		if requestOriginHost != "" {
-			r.Address = requestOriginHost
-		}
-	case *webhook.RegistrationV2:
-		//TODO: do we have any defaults for RegistrationV2 that need to be set?
-		//webhook-schema shows RetryHint, BatchHint, Webhook.SecretHash, and Payload only will have default values
-		//are we setting those values here?
+func (wv webhookValidator) setV1Defaults(r *webhook.RegistrationV1, requestOriginHost string) {
```

Comment:
separated into two separate functions for two reasons:
1. will have different fields being defaulted
2. if we kept as one function and used the `any` interface we would have had to also return an `any` interface and that would have required type switch statement both before, during, and  after the function call